src/guides/gelato-smart-contract-automation/index.md

```diff
@@ -0,0 +1,449 @@
+- [truffle](https://trufflesuite.com/docs/truffle/getting-started/installation)
+
+## Create an Infura account and project
+
+To connect your DApp to the Goerli network, you'll need an Infura account. Sign up for an account [here](https://infura.io/register?utm_source=web3unleashed&utm_medium=webinar&utm_campaign=2022_Sep_gelato-smart-contract-automation_tutorial_content).
+
+Once you're signed in, create a project! Let's call it `gelato-automation`, and select Web3 API from the dropdown
+
+## Register for a MetaMask wallet
+
+To interact with your DApp in the browser, you'll need a MetaMask wallet. You can download it and create one [here](https://metamask.io/download/?utm_source=web3unleashed&utm_medium=webinar&utm_campaign=2022_Sep_gelato-smart-contract-automation_tutorial_content).
+
+## Download VS Code
+
+Feel free to use whatever IDE you want, but we highly recommend using VS Code! You can run through most of this tutorial using the Truffle extension to create, build, and deploy your smart contracts, all without using the CLI! You can read more about it [here](https://trufflesuite.com/blog/build-on-web3-with-truffle-vs-code-extension/).
+
+## Get Some Test Eth
+
+In order to deploy to the public testnets, you'll need some test Eth to cover your gas fees! Make sure you're getting test Eth for either [Goerli](https://goerli-faucet.pk910.de/) or Sepolia.
+
+## Set Up Your Project
+
+Since we're building on our NFT rental marketplace from episode 4, start by cloning the base repository.
+
+```shell
+git clone https://github.com/trufflesuite/unleashed_nft_rental_marketplace.git gelato-demo
+```
+
+We'll only be working with contracts, so hop into the `truffle` directory and install our package dependencies.
+
+```shell
+cd gelato-demo/truffle
+npm i
+```
+
+## Write the automated task
+
+In our marketplace, we will remove listings that are no longer available. A listing becomes unavailable when the end date of the listing has passed. Add this method to `Marketplace.sol`: 
+
+```javascript
+function removeUnavailableListings(Listing[] memory unavailableListings) public {
+    for (uint i = 0; i < unavailableListings.length; i++) {
+        unlistNFT(unavailableListings[i].nftContract, unavailableListings[i].tokenId);
+    }
+}
+```
+
+It steps through a list of unavailable listings and calls our already written `unlistNFT` function, which will remove the NFT from our marketplace's list of listings and decrement the total number of listings available.
+
+## Write `IResolver`
+
+Because our automated task takes in the dynamic argument `unavailableListings` and is dependent on on a few conditions, we'll need to write a resolver. The first thing we will do is create a resolver interface. This isn't explicitly necessary, but is good practice if you have to write multiple resolvers for different smart contracts.
+
+```shell
+truffle create contract IResolver
+```
+Then, paste in the following code:
+
+```javascript
```

Comment:
A bunch of the fenced code blocks incorrectly specify `javascript` rather than `solidity` (assuming it's supported).

Reply:
there's not solidity support, I've just been using javascript cause it's been the closest

Reply:
Ah makes sense